controller/sync_strats/sync_instant.go

```diff
@@ -0,0 +1,528 @@
+
+				state.Sequence = state.Sequence + 1
+				apiSessions = []*edge_ctrl_pb.ApiSession{}
+			}
+		}
+
+		if len(apiSessions) > 0 {
+			state.IsLast = true
+			strategy.sendApiSessionAdded(rtx, true, state, apiSessions)
+		}
+		return nil
+	})
+
+	strategy.ae.GetDbProvider().GetDb().View(func(tx *bbolt.Tx) error {
+		var sessions []*edge_ctrl_pb.Session
+
+		state := &InstantSyncState{
+			Id:       cuid.New(),
+			IsLast:   true,
+			Sequence: 0,
+		}
+
+		for cursor := strategy.ae.GetStores().Session.IterateIds(tx, ast.BoolNodeTrue); cursor.IsValid(); cursor.Next() {
+			current := cursor.Current()
+
+			session, err := strategy.ae.GetStores().Session.LoadOneById(tx, string(current))
+
+			if err != nil {
+				logger.WithError(err).WithField("sessionId", string(current)).Errorf("error querying session [%s]: %v", string(current), err)
+				continue
+			}
+
+			sessionProto, err := sessionToProto(strategy.ae, session)
+
+			if err != nil {
+				logger.WithError(err).WithField("sessionId", string(current)).Errorf("error turning session [%s] into proto: %v", string(current), err)
+				continue
+			}
+
+			sessions = append(sessions, sessionProto)
+
+			if len(sessions) >= chunkSize {
+				state.IsLast = !cursor.IsValid()
+				strategy.sendSessionAdded(rtx, true, state, sessions)
+
+				state.Sequence = state.Sequence + 1
+				sessions = []*edge_ctrl_pb.Session{}
+			}
+		}
+
+		if len(sessions) > 0 {
+			state.IsLast = true
+			strategy.sendSessionAdded(rtx, true, state, sessions)
+		}
+		return nil
+	})
+
+	rtx.Status = env.RouterSyncDone
+}
+
+func (strategy *InstantStrategy) sendApiSessionAdded(rtx *routerTx, isFullState bool, state *InstantSyncState, apiSessions []*edge_ctrl_pb.ApiSession) {
+	stateBytes, _ := json.Marshal(state)
+
+	msgContent := &edge_ctrl_pb.ApiSessionAdded{
+		IsFullState: isFullState,
+		ApiSessions: apiSessions,
+	}
+
+	msgContentBytes, _ := proto.Marshal(msgContent)
+
+	msg := channel2.NewMessage(env.ApiSessionAddedType, msgContentBytes)
+
+	msg.Headers[env.SyncStrategyTypeHeader] = []byte(strategy.Type())
+	msg.Headers[env.SyncStrategyStateHeader] = stateBytes
+
+	rtx.Send(msg)
+}
+
+func (strategy *InstantStrategy) sendSessionAdded(rtx *routerTx, isFullState bool, state *InstantSyncState, sessions []*edge_ctrl_pb.Session) {
+
+	stateBytes, _ := json.Marshal(state)
+
+	msgContent := &edge_ctrl_pb.SessionAdded{
+		IsFullState:  isFullState,
+		Sessions:     sessions,
+	}
+
+	msgContentBytes, _ := proto.Marshal(msgContent)
+
+	msg := channel2.NewMessage(env.SessionAddedType, msgContentBytes)
+	msg.Headers[env.SyncStrategyTypeHeader] = []byte(strategy.Type())
+	msg.Headers[env.SyncStrategyStateHeader] = stateBytes
+	rtx.Send(msg)
+}
+
+type InstantStrategyOptions struct {
+	MaxOutstandingHellos int32
+	MaxConcurrentSyncs   int32
+	RouterTxBufferSize   int
+	HelloSendTimeout     time.Duration
+}
+
+type InstantSyncState struct {
+	Id       string `json:"id"`       //unique id for the sync attempt
```

Comment:
This id is meant to be used to track what sync is currently being worked on. The id is monotonic so they can be compared for "who is newer". This id should be output in log messages to track a sync attempt.